### PR TITLE
Performance metrics

### PR DESCRIPTION
**Pull request by Nat-133**

### CODE REVIEW

File: op2/include/op_lib_core.h [hunk 1]
```diff
@@ -183,7 +183,10 @@ typedef struct {
   float plan_time;  /* time spent in op_plan_get */
   float transfer;   /* bytes of data transfer (used) */
   float transfer2;  /* bytes of data transfer (total) */
-  double mpi_time;   /* time spent in MPI calls */
+  double mpi_time;  /* time spent in MPI calls */
+  #ifdef GPI_ACTIVE
```
Comment (NickWalker1) [inline op2/include/op_lib_core.h]: Is this definitely the flag we're using?

**Comment by NickWalker1** [comment]
Can you check, now that you've moved stuff around, OP2 still builds and validates with MPI? (and gets the same output info for timing)

File: op2/src/common/op_perf_common.cpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      kernel_entry->num_indices++;
+    } else {
+      kernel_entry->comm_info[index]->count += 1;
+      kernel_entry->comm_info[index]->bytes += tot_halo_size;
+    }
+  }
+}
+#endif
+
+#ifdef COMM_PERF
```
Comment (NickWalker1) [inline op2/src/common/op_perf_common.cpp:155]: These lines are unnecessary, it's the same #ifdef as above.

**Comment by NickWalker1** [comment]
With regards to the gpi_time counter. I suppose we will use that identically to the MPI one? As MPI one is just manually incremented after calls to op_timers_core. May be worth making a note of this somewhere?

**Review by NickWalker1 — CHANGES_REQUESTED** [review]
Just nitpicks here! But need to check it all validates & gets the same output as before.

File: op2/src/common/op_perf_common.cpp (hunk 2 above)
Comment (Nat-133) [reply op2/src/common/op_perf_common.cpp:155]: Yeah, I know, but it's how it was done in the original file. I could have changed it, but there might be a reason for it.

File: op2/include/op_lib_core.h (hunk 1 above)
Comment (Nat-133) [reply op2/include/op_lib_core.h]: I've fixed this in the new commit

**Review by NickWalker1 — APPROVED** [review]
Good!

**Pull request merged**